src/app/views/collections/details/CollectionDetailsController.jsx

```diff
@@ -384,19 +397,30 @@ export class CollectionDetailsController extends Component {
     }
 
     handleCollectionPageDeleteClick(deletedPage, state) {
```

Comment:
This function is a monolithic 96 liner. I think this needs to be split up, might be worth creating a 'detailsDeletionController' or something, then taking the sub-functions within this function and some others from this class and adding them to it.